lib/doorkeeper/oauth/helpers/scope_checker.rb

```diff
@@ -7,16 +7,21 @@ module ScopeChecker
         class Validator
           attr_reader :parsed_scopes, :scope_str
 
-          def initialize(scope_str, server_scopes, application_scopes)
+          def initialize(grant_type, scope_str, server_scopes, application_scopes)
```

Comment:
I think it would be better to refactor the initializer to use kwargs as it makes it easier to understand what arguments / params do we use.